### PR TITLE
Optimize Scene Loading

### PR DESCRIPTION
**Pull request by HexaField**
## Summary

- [x] Authorization no longer blocking first paint for location loading
- [x] `userID` is now in EngineState and reactive, though most references are still non-reactive
- [x] `location` service now has a specific query that allows unauthenticated requests in order to load scenes quicker
- [x] `location` service now resolves a string `sceneURL` property instead of a `StaticResourceType` `sceneAsset`
- [x] Added API server debug menu
- [x] Minimized the number of API requests prior to loading a location (should be at a minimum 2 instead of like a dozen, though likely more as many things are handled in parallel)
- [x] Optimized cross imports of location and editor
- [x] Lazily load corto worker and restrict number from 2 to 1
- [x] Disable broken and unused AvatarUISystem
- [x] Simplify a variety of states
- [x] Add stacktracing to API requests
- [x] Stop re-initializing KTX2 loader
- [x] Lazily load noise textures for SDF
- [x] Disable visual script initializing an instance of all components
- [x] Move logic to ensure user has an avatar to backend under app login event

## References
closes #_insert number here_

### CODE REVIEW

File: packages/spatial/src/EngineState.ts
Comment (the-reality-engineer) [inline packages/spatial/src/EngineState.ts:1]: Good

File: packages/client-core/src/components/Debug/APIDebug.tsx
Comment (the-reality-engineer) [inline packages/client-core/src/components/Debug/APIDebug.tsx:1]: Awesome!

**Comment by the-reality-engineer** [comment]
Looks like there are some minor linting and ts errors. About to pull this down and try it out. Nice work!

**Comment by DanielBelmes** [comment]
Seeing this error a lot on this branch. No visible defects from it
image

**Comment by DanielBelmes** [comment]
Avatar preview mode is not working on this branch

**Comment by HexaField** [comment]
> Avatar preview mode is not working on this branch

fixed

**Review by the-reality-engineer — APPROVED** [review]

**Pull request merged**